homeassistant/components/device_automation/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""Helpers for device automations."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import split_entity_id
+from homeassistant.helpers.entity_registry import async_entries_for_device
+from homeassistant.loader import async_get_integration, IntegrationNotFound
+
+DOMAIN = 'device_automation'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up device automation."""
+    hass.components.websocket_api.async_register_command(
+        websocket_device_automation_list_triggers)
+    return True
+
+
+async def async_get_device_automation_triggers(hass, device_id):
+    """List device triggers."""
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    domains = set()
+    triggers = []
+    device = device_registry.async_get(device_id)
+    for entry_id in device.config_entries:
+        config_entry = hass.config_entries.async_get_entry(entry_id)
+        domains.add(config_entry.domain)
+
+    entities = async_entries_for_device(entity_registry, device_id)
+    for entity in entities:
+        domains.add(split_entity_id(entity.entity_id)[0])
+
+    for domain in domains:
+        integration = None
+        try:
+            integration = await async_get_integration(hass, domain)
+        except IntegrationNotFound:
+            _LOGGER.exception('Integration %s not found', domain)
```

Comment:
We should never use `exception` unless it's an unexpected exception. This one we know why it happens (but shouldn't). I suggest we log a `warning`

Reply:
Fixed.